lib/componentmanager.js

```diff
@@ -247,6 +265,16 @@
 
         return componentId;
     };
+
+    /**
+     * Add a component object
+     */
+    ComponentManager.prototype.addMetaComponent = function (component) {
+        var componentId = this.getComponentId();
+        component.id = componentId;
+
+        this._allComponents[componentId] = component;
```

Comment:
Should this also track just meta-components? It looks like allComponents could be a mix of Layer and Meta components if they are both allowed to load on startup.